server/memo.go

```diff
@@ -30,6 +30,16 @@ func (s *Server) registerMemoRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusBadRequest, "Malformatted post memo request").SetInternal(err)
 		}
 
+		// Find system settings
+		systemSettingKey := api.SystemSettingDisablePublicMemosName
+		systemSetting, err := s.Store.FindSystemSetting(ctx, &api.SystemSettingFind{
```

Comment:
This can probably be solved with another ``if err != nil`` check?

Reply:
You're right. Totally forgot to do an early return here :D